server/src/copilot/utils/constants.ts

```diff
@@ -0,0 +1,35 @@
+export const REGEX = {
+    COMMENT: /\/\* Analyze the following code in .*? and complete the code based on the context. .*? Remove this commented instruction in complete code. \*\//g,
+    COMMENTED: /\/\/.*$|\/\*[\s\S]*?\*\//gm,
+    BLOCK_START: /{\s*$/,
+    BLOCK_END: /^\s*}\s*$/,
+    EMPTY_LINE: /^\s*$/,
+    TRIPLE_QUOTES_CONTENT: /```([\s\S]*?)```/,
+    KEYWORDS: /\b(namespace|import|contract|transaction|public field|from|extends|asset|concept|template)\b/g,
+    GENERATOR_COMMENT: "// This code is generated by Accord Copilot Model Generator"
+};
+
+export const PROVIDERS = {
```

Comment:
This sort of stuff should be retrieved from the LLM interface not global and scattered around.

Reply:
Fixed! Now, each language model provider encapsulates its own configuration details and endpoint URLs within its class.
eg: getIdentifier(): string {
        return 'openai';
    }